crates/types/src/motion_file.rs

```diff
@@ -3,13 +3,14 @@ use std::{fs::File, path::Path, time::Duration};
 use color_eyre::eyre::{Result, WrapErr};
 use serde::{Deserialize, Deserializer, Serialize, Serializer};
 use serde_json::from_reader;
+use splines::{impl_Interpolate, Interpolation, Key, Spline};
 
-use crate::{Joints, LinearInterpolator};
+use crate::Joints;
 
 #[derive(Clone, Debug, Default, Deserialize, Serialize)]
 pub struct MotionFile {
-    initial_positions: Joints,
-    frames: Vec<MotionFileFrame>,
+    pub initial_positions: Joints,
+    pub frames: Vec<MotionFileFrame>,
```

Comment:
Is it really needed to make them `pub`? Same for the `MotionFileFrame`?

Reply:
Despite creation, these structs do not contain any methods and are only used to be converted into interpolators.
Is it okay to use `pub` then?

Reply:
Yes, then it is okay.